configstack/module.go

```diff
@@ -83,7 +84,18 @@ func resolveTerraformModule(terragruntConfigPath string, terragruntOptions *opti
 		return nil, err
 	}
 
-	return &TerraformModule{Path: modulePath, Config: *terragruntConfig, TerragruntOptions: opts}, nil
+	if terragruntConfig.Terraform != nil {
+		s := []string{filepath.Dir(terragruntConfigPath), "*.tf"}
+		matches, err := filepath.Glob(strings.Join(s, ""))
+		if err != nil {
+			return nil, err
+		}
+		if terragruntConfig.Terraform.Source != "" || matches != nil {
```

Comment:
Looks like Go does use short-circuit evaluation on its Boolean operators, the new logic wraps the nil, nil return rather than the primary return and should look much cleaner!